### PR TITLE
Support additional security groups in machinepool creation

### PR DESCRIPTION
**Pull request by xueli181114**
Support additional security groups in machinepool creation

### CODE REVIEW

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.14-nightly-x86-conc-builds-3nodes

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa_hcp-4.14-nightly-x86-data-path-9nodes

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa_hcp-4.15-ec-x86-data-path-9nodes

File: ci-operator/step-registry/rosa/conf/machine-pool/rosa-conf-machine-pool-ref.yaml [hunk 1]
```diff
@@ -62,6 +62,9 @@ ref:
   - name: MP_TIMEOUT
     default: "900"
     documentation: Set to number of seconds for waiting the htpasswd idp is ready to login.
+  - name: MP_ADDITIONAL_SECURITY_GROUP
+    default: "false"
+    documentation: "Set the additional security groups for the machinepool"
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/conf/machine-pool/rosa-conf-machine-pool-ref.yaml:67]: It is not `set`, it should be 'Enable'.

File: ci-operator/step-registry/rosa/conf/machine-pool/rosa-conf-machine-pool-commands.sh [hunk 2]
```diff
@@ -53,7 +53,11 @@ fi
 
 TAINTS="prowci=true:NoSchedule"
 if [[ ! -z "$MP_TAINTS" ]]; then
-  TAINTS="${MP_TAINTS}"
+  if [[ $MP_TAINTS = NONE ]];then
+    TAINTS=""
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply ci-operator/step-registry/rosa/conf/machine-pool/rosa-conf-machine-pool-commands.sh:57]: @yasun1 I believe there are some OCP jobs are using the local zone machinepool which must have the default taints. I don't want to affect others job.

File: ci-operator/step-registry/rosa/conf/machine-pool/rosa-conf-machine-pool-commands.sh [hunk 3]
```diff
@@ -156,4 +174,28 @@ else
   done
 fi
 
+# Watch the nodes started
+if [[ "$HOSTED_CP" == "true" ]]; then
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply ci-operator/step-registry/rosa/conf/machine-pool/rosa-conf-machine-pool-commands.sh:178]: @yasun1 Is it working for HCP?

File: ci-operator/step-registry/rosa/cluster/wait-ready/nodes/rosa-cluster-wait-ready-nodes-commands.sh [hunk 4]
```diff
@@ -150,6 +150,19 @@ function getDesiredComputeCount {
...
     desired_compute_count=$(rosa describe cluster -c "$CLUSTER_ID"  -o json  |jq -r '.nodes.compute')
   fi
+  if [[ ! -z ${MP_NAME} ]];then
+    compute_count=$(rosa describe machinepool ${MP_NAME}  -c "$CLUSTER_ID"  -o json  |jq -r '.replicas')
+      if [[ "$compute_count" = "null" ]]; then 
+        echo "--auto-scaling enabled, retrieving min_replica count desired"
+        if [[ $HOSTED_CP = "true" ]];then
+          compute_count=$(rosa describe machinepool  ${MP_NAME} -c "$CLUSTER_ID" -o json  | jq -r '.autoscaling.min_replica') 
+        else
+          compute_count=$(rosa describe cluster -c "$CLUSTER_ID" -o json  | jq -r '.autoscaling.min_replicas*')
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/cluster/wait-ready/nodes/rosa-cluster-wait-ready-nodes-commands.sh:160]: Is it right to describe the cluster? And there could be a compile error to use '*' in jq.

File: ci-operator/step-registry/rosa/cluster/wait-ready/nodes/rosa-cluster-wait-ready-nodes-commands.sh [hunk 5]
```diff
@@ -150,6 +150,19 @@ function getDesiredComputeCount {
     echo "--auto-scaling disabled, --replicas set, retrieving replica count desired"
     desired_compute_count=$(rosa describe cluster -c "$CLUSTER_ID"  -o json  |jq -r '.nodes.compute')
   fi
+  if [[ ! -z ${MP_NAME} ]];then
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/cluster/wait-ready/nodes/rosa-cluster-wait-ready-nodes-commands.sh:153]: The variable is not defined. Need to update the machinepool step, and get it from there.

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa-4.12-nightly-x86-cluster-density-v2-3nodes

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-sts-advanced-e2e-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-sts-advanced-e2e-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-sts-advanced-e2e-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-sts-advanced-e2e-f3

**Comment by xueli181114** [comment]
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/49628/rehearse-49628-periodic-ci-openshift-openshift-tests-private-master-rosacli-aws-rosa-sts-advanced-e2e-f3/1766673431551545344 provision steps passed

**Comment by xueli181114** [comment]
@yasun1 Please help to review.

**Pull request merged**